VC/instrumentation/read_labjack.py

```diff
@@ -0,0 +1,268 @@
+    configures a different setting:
+
+      Set bit 4-5 = GainIndex:
+        GainIndex   Gain   Max V    Min V
+        b00         1      10.1     -10.6
+        b01         10     1.01     -1.06
+        b10         100    0.101    -0.106
+        b11         1000   0.0101   -0.0106
+    
+      Set bit 7 for differential reading.
+
+    Ex:
+      Channel 0 single ended with gain 10:  (0, SING | X10)
+      Channel 0,1 differential with gain 1: (0, DIFF | X1)
+
+'''
+
+######### BEGIN USER ADJUSTABLE #########
+
+# Stream settings
+scan_frequency   = 25
+resolution_index = 2
+settling_factor  = 2
+samples_per_packet = 50
+channel_settings = [(86, DIFF | X1000), # L_RUN_TANK (SEEMS GOOD. CHECK CAL)
+                    (87, DIFF | X1000), # L_THRUST (VERY NOISY)
+
+                    (82, DIFF | X100),  # P_INJECTOR (GOOD!)
+                    (85, DIFF | X100),  # P_COMB_CHMBR
+                    (84, DIFF | X100),  # P_N2O_FLOW
+                    (83, DIFF | X100),  # P_N2_FLOW
+                    (81, DIFF | X100),  # P_RUN_TANK
+                    
+                    (55, SING | X100),  # T_COMB_CHMBR
+                    (53, SING | X100),  # T_POST_COMB_CHMBR
+                    (57, SING | X100),  # T_INJECTOR
+                    (49, SING | X1000)]  # T_RUN_TANK 
+
+# These gains and offsets convert raw volts to the SI unit of the sensor
+
+# 6895 converts PSI to Pa. *2 since running on 5V supply rather than 10V
+GAIN_P_COMB_CHMBR = 2*6895*0.0       # [Pa/V] at 10V supply TODO: Not in use
+GAIN_P_N2O_FLOW   = 2*6895*9982.03   # [Pa/V] at 10V supply 
+GAIN_P_N2_FLOW    = 2*6895*9936.41   # [Pa/V] at 10V supply
+GAIN_P_RUN_TANK   = 2*6895*10206.74  # [Pa/V] at 10V supply
+GAIN_P_INJECTOR   = 2*6895*10170.95  # [Pa/V] at 10V supply
+
+GAIN_L_RUN_TANK   = 2014.22   # [N/V]  at 5V supply
+GAIN_L_THRUST     = 466000    # [N/V]  at 5V supply 
+
+OFFSET_L_THRUST   = -25.8     # [N] # TODO double check
+OFFSET_L_RUN_TANK = -5.8      # [N] # TODO add 
+
+# Channel mapping from LJ to sensor
+CHAN_L_RUN_TANK   = 'AIN86'
+CHAN_L_THRUST     = 'AIN87'
+
+CHAN_P_COMB_CHMBR = 'AIN85'
+CHAN_P_N2O_FLOW   = 'AIN84'
+CHAN_P_N2_FLOW    = 'AIN83'
+CHAN_P_RUN_TANK   = 'AIN81'
+CHAN_P_INJECTOR   = 'AIN82'
+
+CHAN_T_RUN_TANK   = 'AIN49'
+CHAN_T_INJECTOR   = 'AIN57'
+CHAN_T_COMB_CHMBR = 'AIN55'
+CHAN_T_POST_COMB  = 'AIN53'
+
+CHAN_SHUNT        = 'AIN82' # Not implemented yet on cart
+
+#########  END USER ADJUSTABLE  #########
+
+# Set up the stream
+d = u6.U6()
+d.streamConfig(
+        ScanFrequency   = scan_frequency,
+        ChannelNumbers  = [x[0] for x in channel_settings],
+        ChannelOptions  = [x[1] for x in channel_settings],
+        NumChannels     = len(channel_settings),
+        ResolutionIndex = resolution_index,
+        SettlingFactor  = settling_factor,
+        SamplesPerPacket = samples_per_packet)
+
+# Avoid having to power cycle the LJ on restart
+try:
+    d.streamStop()
+except:
+    pass
```

Comment:
This is just for hotplug support during testing. Otherwise if you ctrl+c the program while connected, you have to unplug and replug the LJ before you can connect again. I guess it's probably still good to include tho?